textx/exceptions.py

```diff
@@ -1,14 +1,22 @@
 # TextX Exceptions
 class TextXError(Exception):
-    def __init__(self, message, line=None, col=None):
+    def __init__(self, message, line=None, col=None, err_type=None):
         super(TextXError, self).__init__(message.encode('utf-8'))
         self.line = line
         self.col = col
+        self.err_type = err_type
 
 
 class TextXSemanticError(TextXError):
-    pass
+    def __init__(self, message, line=None, col=None, err_type=None, expected_obj_cls=None):
```

Comment:
There are PEP-8 violation at multiple places. Try to enable linter in your editor.
If you have trouble let me know and I'll fix it afterwards.